echo_server.py

```diff
@@ -21,10 +24,49 @@ def echo_server():
                 if len(response) < 32:
                     done = True
                 text_received = "{}{}".format(text_received, response)
-            conn.sendall(text_received)
+            if text_received:
+                response_user = parase_request(text_received)
+            conn.sendall(response_user)
     except KeyboardInterrupt:
        server_socket.close()
 
+def response_ok(uri):
+    first_line = 'HTTP/1.1 200 OK'
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header = 'Content-Type: text/plain'
+    body = uri
+    size = str(sys.getsizeof(body))
+    end = '\r\n'
+    response  =  [first_line, timestamp, header, size, body, end]
+    return end.join(response)
+
+def response_error(error_code, error_message):
+    first_line = 'HTTP/1.1 {} {}'.format(error_code, error_message)
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header = 'Content-Type: text/plain'
+    body = '{} {}'.format(error_code, error_message)
+    size = str(sys.getsizeof(body))
+    end = '\r\n'
+    error_response = [first_line, timestamp, header, size, body, end]
```

Comment:
Again, the empty line comes to late to show the body and size as elements in your browser window.  If you are seeing them there, it's an anomaly caused by the browser you're using and the forgiving nature of browsers.  On the other hand, your first line is build perfectly.  Good work there.